File: testprojects/scala_main/pipelines/DONOT_OPEN_SCALA/code/src/main/scala/org/main/scla_dep_mgmt/graph/src_csv_special_char_column_name.scala
package org.main.scla_dep_mgmt.graph

import io.prophecy.libs._
import org.main.scla_dep_mgmt.config.ConfigStore._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import java.time._

object src_csv_special_char_column_name {

  def apply(spark: SparkSession): DataFrame =
    spark.read
      .format("csv")
      .option("header", true)
      .option("sep",    ",")
      .schema(
        StructType(
          Array(
            StructField("c   short  --",     StringType, true),
            StructField("c-int-column type", StringType, true),
            StructField("-- c-long",         StringType, true),
            StructField("c-decimal",         StringType, true),
            StructField("c  float",          StringType, true),
            StructField("c--boolean",        StringType, true),
            StructField("c- - -double",      StringType, true),
            StructField("c___-- string",     StringType, true),
            StructField("c  date",           StringType, true),
            StructField("c_timestamp",       StringType, true)
          )
        )
      )
      .load("dbfs:/Prophecy/qa_data/csv/special_char_column_name")
      .cache()

}

File: testprojects/master_python_mainone/pipelines/PythonDoAnything/code/pythondoanything/graph/src_csv_special_char_column_name.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.libs import typed_lit
from pythondoanything.config.ConfigStore import *
from pythondoanything.udfs.UDFs import *

def src_csv_special_char_column_name(spark: SparkSession) -> DataFrame:
    return spark.read\
        .schema(
          StructType([
            StructField("c   short  --", StringType(), True), StructField("c-int-column type", StringType(), True), StructField("-- c-long", StringType(), True), StructField("c-decimal", StringType(), True), StructField("c  float", StringType(), True), StructField("c--boolean", StringType(), True), StructField("c- - -double", StringType(), True), StructField("c___-- string", StringType(), True), StructField("c  date", StringType(), True), StructField("c_timestamp", StringType(), True)
        ])
        )\
        .option("header", True)\
        .option("sep", ",")\
        .csv("dbfs:/Prophecy/qa_data/csv/special_char_column_name")
